azurerm/internal/services/web/function_app_resource.go

```diff
@@ -399,6 +417,8 @@ func resourceFunctionAppUpdate(d *schema.ResourceData, meta interface{}) error {
 	appServicePlanID := d.Get("app_service_plan_id").(string)
 	enabled := d.Get("enabled").(bool)
 	clientAffinityEnabled := d.Get("client_affinity_enabled").(bool)
+	clientCertMode := d.Get("client_cert_mode").(string)
+	clientCertEnabled := clientCertMode != "Ignore"
```

Comment:
meaning this could become:

```suggestion
	clientCertEnabled := clientCertMode != ""
```